model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionEntity.java

```diff
@@ -0,0 +1,152 @@
+import org.keycloak.models.map.storage.hotRod.common.HotRodStringPair;
+import org.keycloak.models.map.storage.hotRod.common.UpdatableHotRodEntityDelegateImpl;
+import org.keycloak.models.map.userSession.MapUserSessionEntity;
+
+import java.util.Objects;
+import java.util.Set;
+
+@GenerateHotRodEntityImplementation(
+        implementInterface = "org.keycloak.models.map.userSession.MapUserSessionEntity",
+        inherits = "org.keycloak.models.map.storage.hotRod.userSession.HotRodUserSessionEntity.AbstractHotRodUserSessionEntityDelegate"
+)
+@ProtoDoc("@Indexed")
+public class HotRodUserSessionEntity extends AbstractHotRodEntity {
+
+    @ProtoField(number = 1, required = true)
+    public int entityVersion = 1;
+
+    @ProtoField(number = 2, required = true)
+    public String id;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 3)
+    public String realmId;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 4)
+    public String userId;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 5)
+    public String brokerSessionId;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 6)
+    public String brokerUserId;
+
+    @ProtoField(number = 7)
+    public String loginUsername;
+
+    @ProtoField(number = 8)
+    public String ipAddress;
+
+    @ProtoField(number = 9)
+    public String authMethod;
+
+    @ProtoField(number = 10)
+    public Boolean rememberMe;
+
+    @ProtoField(number = 11)
+    public Integer started;
+
+    @ProtoField(number = 12)
+    public Integer lastSessionRefresh;
+
+    @ProtoField(number = 13)
+    public Long expiration;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 14)
+    public Set<HotRodStringPair> notes;
+
+    @ProtoField(number = 15)
+    public HotRodSessionState.State state;
+
+    @ProtoField(number = 16)
+    public HotRodSessionState.PersistenceState persistenceState;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 17)
+    public Set<HotRodStringPair> authenticatedClientSessions;
+
+    @ProtoField(number = 18)
+    public Boolean offline;
+
+    public static abstract class AbstractHotRodUserSessionEntityDelegate extends UpdatableHotRodEntityDelegateImpl<HotRodUserSessionEntity> implements MapUserSessionEntity {
+
+        @Override
+        public String getId() {
+            return getHotRodEntity().id;
+        }
+
+        @Override
+        public void setId(String id) {
+            HotRodUserSessionEntity entity = getHotRodEntity();
+            if (entity.id != null) throw new IllegalStateException("Id cannot be changed");
+            entity.id = id;
+            entity.updated |= id != null;
+        }
+
+        @Override
+        public UserSessionModel.State getState() {
```

Comment:
It should be possible to implement these methods by the generator. Previously, it was not possible because we stored these in the `Map<String, Enum>`, however, it should work with a simple field.

Reply:
done